tools/topology/sof/pipe-eq-capture.m4

```diff
@@ -35,7 +32,7 @@ C_CONTROLMIXER(Master Capture Volume, PIPELINE_ID,
 	Channel register and shift for Front Left/Right,
 	LIST(`	', KCONTROL_CHANNEL(FL, 1, 0), KCONTROL_CHANNEL(FR, 1, 1)))
 
-define(`CONTROL_NAME', `CONTROL_NAME_SWITCH')
+ifdef(`CONTROL_NAME_SWITCH', `define(`CONTROL_NAME', `CONTROL_NAME_SWITCH')', `undefine(`CONTROL_NAME')')
```

Comment:
@juimonen why is this even needed? I dont see the macro "CONTROL_NAME" used at all in this m4 file?

Reply:
@ranj063 CONTROL_NAME is used in mixercontrol.m4. If you don't define/redifine it, control will be named with the same name always.

Reply:
@juimonen this is too confusing. I think the proper way to fix the problem is to remove the CONTROL_NAME macro usage from mixercontrol.m4 and pass the name to the C_CONTROLMIXER macro.

The name to be passed should be set to CONTROL_NAME_SWITCH/VOLUME if defined and if not define them here with the right name in this m4 file instead of passing macro to the next level. 

That way, your changes in lines 83/84 won't be needed.

Reply:
@ranj063 sorry now I'm lost, can you give me example what you mean?